mm2src/coins/z_coin/storage/z_change_notes/sqlite.rs

```diff
@@ -0,0 +1,133 @@
+use super::{ChangeNote, ChangeNoteStorage, ChangeNoteStorageError};
+
+use db_common::async_sql_conn::AsyncConnError;
+use db_common::sqlite::rusqlite::params;
+use db_common::sqlite::{run_optimization_pragmas, validate_table_name};
+use itertools::Itertools;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+
+fn table_name(addr: &str) -> String { format!("{addr}_change_notes_cache") }
+
+fn create_change_note_table(for_addr: &str) -> Result<String, AsyncConnError> {
+    let table_name = table_name(for_addr);
+    validate_table_name(&table_name)?;
```

Comment:
not a blocker: Can we perform the validation directly in `table_name`? If we forget calling `validate_table_name` (which is quite possible), it will be security concern.